src/views/docs-view/types.ts

```diff
@@ -40,6 +42,21 @@ export interface DocsViewProps {
 	 * An optional, human-readable name to be rendered where the section is referenced as a versioned project
 	 */
 	projectName?: string
+
+	/**
+	 * Product data. Note we include `currentRootDocsPath` for version switching.
+	 */
+	product: ProductWithCurrentRootDocsPath
```

Comment:
🚧 This isn't quite accurate. `DocsView` doesn't use a `product` prop, this is actually for the `CurrentProductProvider` that's initialized in `src/pages/_app.tsx`. It's value is pulled from `pageProps` here:

https://github.com/hashicorp/dev-portal/blob/1f3bac6d9a870d0e1f9d7a30e6a5c84af21b97d3/src/pages/_app.tsx#L97